tests/phpunit/integration/Modules/Analytics_4/Synchronize_PropertyTest.php

```diff
@@ -0,0 +1,225 @@
+		);
+	}
+
+	public function set_analytics_settings() {
+		// Ensure Anlytics 4 module is connected
+		$this->options->set(
+			Analytics_4_Settings::OPTION,
+			array(
+				'accountID'       => '12345678',
+				'propertyID'      => '987654321',
+				'webDataStreamID' => '1234567890',
+				'measurementID'   => 'G-A1B2C3D4E5',
+			)
+		);
+
+		// Set owner ID, using base analytics settings.
+		$this->options->set(
+			Settings::OPTION,
+			array(
+				'ownerID' => get_current_user_id(),
+			)
+		);
+	}
+
+	public function fake_analytics_4_property_object( $property_id, $create_time ) {
+		FakeHttp::fake_google_http_handler(
+			$this->analytics_4->get_client(),
+			function ( Request $request ) use ( $property_id, $create_time ) {
+				$url = parse_url( $request->getUri() );
+
+				$property = new GoogleAnalyticsAdminV1betaProperty();
+				$property->setCreateTime( $create_time );
+				$property->setName( "properties/{$property_id}" );
+
+				if ( "/v1beta/properties/{$property_id}" === $url['path'] ) {
+					return new Response(
+						200,
+						array(),
+						json_encode(
+							$property
+						)
+					);
+				}
+
+				return new Response( 200 );
+			}
+		);
+	}
+
+	public function test_register() {
+		remove_all_actions( Synchronize_Property::CRON_SYNCHRONIZE_PROPERTY );
+
+		$this->synchronize_property->register();
+		$this->synchronize_property->maybe_schedule_synchronize_property();
+
+		$this->assertTrue( has_action( Synchronize_Property::CRON_SYNCHRONIZE_PROPERTY ) );
+
+		$this->assertTrue(
+			(bool) wp_next_scheduled( Synchronize_Property::CRON_SYNCHRONIZE_PROPERTY )
+		);
+	}
+
+	public function test_schedule_synchronize_property__cron_callback() {
+		$property_id    = '987654321';
+		$create_time    = '2014-10-02T15:01:23Z';
+		$create_time_ms = Synchronize_Property::convert_time_to_unix_ms( $create_time );
+
+		$this->authentication->get_oauth_client()->set_granted_scopes(
+			$this->analytics_4->get_scopes()
+		);
+
+		$this->fake_analytics_4_property_object( $property_id, $create_time );
+
+		// Confirm that user is authenticated and module is connected, as it will be needed in the cron callback.
+		$this->assertTrue( $this->authentication->is_authenticated() );
+		$this->assertTrue( $this->analytics_4->is_connected() );
+
+		$this->synchronize_property->register();
+
+		$settings             = $this->analytics_4->get_settings()->get();
+		$property_create_time = $settings['propertyCreateTime'];
+
+		$this->assertNotEquals( $create_time_ms, $property_create_time );
+
+		// Invoke cron callback function.
+		do_action( Synchronize_Property::CRON_SYNCHRONIZE_PROPERTY );
+
+		$settings             = $this->analytics_4->get_settings()->get();
+		$property_create_time = $settings['propertyCreateTime'];
+		// Property create time should be synced from the retrieved property object.
+		$this->assertEquals( $create_time_ms, $property_create_time );
+	}
+
+	public function test_schedule_synchronize_property__create_time_has_value() {
```

Comment:
This seems like an older version of the test (looking at the name) but now that this method only schedules the sync, it doesn't make sense to make assertions about the setting's value not changing because the cron never runs.